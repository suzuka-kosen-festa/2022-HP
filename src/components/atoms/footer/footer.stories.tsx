import { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { Footer } from ".";
type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default { component: Footer } as ComponentMeta<T>;

export const Default: Story = {};
