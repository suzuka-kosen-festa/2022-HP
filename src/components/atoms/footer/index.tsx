import React from "react";
import tw from "twin.macro";
import { Text } from "../text";
const Footer: React.FC = () => (
  <footer css={tw` w-screen`}>
    <Text css={tw`bg-black text-center p-2`}>&copy; 第57回鈴鹿高専祭実行委員会</Text>
  </footer>
);

export { Footer };
