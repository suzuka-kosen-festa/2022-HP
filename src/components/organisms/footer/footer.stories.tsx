import { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { Footer } from ".";

type Story = ComponentStoryObj<typeof Footer>;

export default { component: Footer } as ComponentMeta<typeof Footer>;

export const Default: Story = {};
