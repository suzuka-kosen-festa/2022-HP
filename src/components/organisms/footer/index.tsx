import React from "react";
import tw from "twin.macro";
import { AnchorText } from "../../atoms/anchor-text";

const Footer: React.FC = () => (
  <footer role={"contentinfo"} css={tw` w-screen`}>
    <AnchorText
      href="https://github.com/suzuka-kosen-festa/2022-HP/blob/main/LICENSE"
      css={tw`bg-black text-center p-2`}
    >
      &copy; 第57回鈴鹿高専祭実行委員会
    </AnchorText>
  </footer>
);

export { Footer };
