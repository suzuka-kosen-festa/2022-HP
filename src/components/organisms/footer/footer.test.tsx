import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./footer.stories";

const { Default } = composeStories(stories);

const footeroptions = {
  name: "",
};

const linkoptions = {
  name: "Â© This is test",
};

describe("(components) organisms/footer", () => {
  test("to be organisms", () => {
    const { container } = render(<Default />);
    expect(container).toBeOrganism();
  });
  test("to be [role=contentinfo]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("contentinfo", footeroptions)).toBeInTheDocument();
  });
  test("to be [role=link]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("link", linkoptions)).toBeInTheDocument();
  });
  test("take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
