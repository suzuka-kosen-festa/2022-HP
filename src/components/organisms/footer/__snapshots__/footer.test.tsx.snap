// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(components) organisms/footer take snap shot 1`] = `
<div>
  <footer
    class="twin-fas8s8-Footer"
    role="contentinfo"
  >
    <a
      class="twin-1msmckp-AnchorText-Footer ey6ijqd0"
      href="https://github.com/suzuka-kosen-festa/2022-HP/blob/main/LICENSE"
    >
      © 第57回鈴鹿高専祭実行委員会
    </a>
  </footer>
</div>
`;
